### ISSUE TITLE
Add kubernetes chart

### ISSUE THREAD

**Issue by tuna2134** [bluesky-social/pds#57]
I want to host pds at kubernetes.

**Comment by temujin9** [comment]
See #3 for prior responses to this idea.

**Comment by temujin9** [comment]
I'm considering providing this in a separate repo, as you're not the only other interested. If so, I will mention it in that thread.

**Comment by tuna2134** [comment]
Ok thanks

**Comment by Nerkho** [comment]
I started working on a Helm chart.

It seems to work fine for me so far. Only have issue with the handle validation via HTTP but validation via DNS works fine.

It would be great if you can also test this and provide some feedback : https://github.com/Nerkho/bluesky-pds-helm-chart
(feel free to fork it, create PRs, etc...)

**Comment by bnewbold** [comment]
Just to confirm: we aren't planning to develop and maintain additional deployment tooling, including kubernetes, in the near future. But we fully support folks collaborating and sharing tooling around this! If there are tiny and low-risk changes to the service itself to un-block this, let us know; it runs fine in containerization with config via env vars, so should be pretty straight-forward to support in other systems.

**Comment by Nerkho** [comment]
Just a small update. I have moved the chart to another repo to facilitate publishing. I've also published the chart, so it's now usable without cloning the whole GitHub repository.

All is here : https://github.com/Nerkho/helm-charts/tree/main/charts/bluesky-pds

**Comment by temujin9** [comment]
@Nerkho thank you for picking this up. I'll try it out on my local machine shortly.

**Comment by hynet-mel** [comment]
> All is here : https://github.com/Nerkho/helm-charts/tree/main/charts/bluesky-pds

@Nerkho Apparently that chart is no longer available, resulting in a 404 sadly

**Comment by Nerkho** [comment]
> > All is here : https://github.com/Nerkho/helm-charts/tree/main/charts/bluesky-pds
> 
> @Nerkho Apparently that chart is no longer available, resulting in a 404 sadly

Yes, sorry, meanwhile I moved some of my stuff to codeberg. The bluesky-pds chart is [here](https://codeberg.org/nerkho/helm-charts/src/branch/main/charts/bluesky-pds)

But I might bring this repo back to GitHub very soon.

**Comment by madmalkav** [comment]
Anyone willing to write a guide on how to install PDS on Kubernetes for dummies, including ingress controller and cert-manager? I'm willing to add some cash to a bounty if needed.

**Comment by vcarl** [comment]
Also quite interested in this. I'd love to host and I'm currently administering a k8s cluster, [the account migration warnings](https://github.com/bluesky-social/pds/blob/main/ACCOUNT_MIGRATION.md) make me nervous about setting up a PDS — I'd love to avoid overpaying for a large instance before there's usage, but knowing it's difficult to migrate later on makes me uncomfortable with the risk of under-sizing.

**Comment by Nerkho** [comment]
@madmalkav @vcarl I now have a blog post up that explains how to deploy the Helm chart above in more details.

https://nerkho.ch/blog/self-hosted-pds-on-k8s/

I don't go deep into the cert-manager and ingress controller part, but it should help to get started.

**Comment by axfelix** [comment]
Hi folks -- thanks for these contributions! I am considering opening up another repo where users can contribute helm charts and other custom PDS deploy scripts. Would that be of interest?
